tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -129,15 +122,28 @@ bool IsHLSLNumericUserDefinedType(clang::QualType type) {
   return false;
 }
 
+bool IsHLSLNumericAggregate(clang::QualType QT) {
```

Comment:
I don't think the name captures the right intent.

The purpose in uses of this function is to verify that we have a struct or class type (record type that will map to a struct type in llvm to which we can associate type annotation metadata) that only contains numeric types (isn't and doesn't contain a special HLSL object type).  Matrix, vector, and array types should not be included as the top-level type, but are fine as members of the top-level type.

Maybe `IsHLSLNumericRecordType` or `IsHLSLNumericStructOrClassType` would be more appropriate?

Reply:
That naming doesn't quite work either. Vector and Matrix types are Record types in DXC too.